src/config.ml

```diff
@@ -136,12 +136,56 @@ module Conf_map = struct
 
   include Gmap.Make(K)
 
+  let is_valid_config t =
+    let ensure_mem k err = if mem k t then Ok () else Error err in
+    (* let ensure_not k err = if not (mem k t) then Ok () else Error err in *)
+    let open Rresult in
+    R.reword_error (fun err -> `Msg ("not a valid config: " ^  err))
+      ( ensure_mem Cipher "config must specify 'cipher AES-256-CBC'"
+        >>= fun () ->
+        (if mem Cipher t && get Cipher t <> "AES-256-CBC" then
+           Error "currently only supported Cipher is 'AES-256-CBC'"
+         else Ok ()) 
+      )
+
+
+  let is_valid_server_config t =
+    let ensure_mem k err = if mem k t then Ok () else Error err in
+    let ensure_not k err = if not (mem k t) then Ok () else Error err in
+    let open Rresult in
+    R.reword_error (fun err -> `Msg ("not a valid server config: " ^  err))
+      ( 
+        match is_valid_config t with
+          | Error _ -> Error "Not a valid config"
+          | Ok _ -> Ok () >>= fun () ->
```

Comment:
This needs to go before the `R.reword_error`, unless we get
`not a valid server config: not a valid config: currently only supported Cipher is 'AES-256-CBC'` which is a tad verbose :)

(same in the client config checker)